compiler/AST/iterator.cpp

```diff
@@ -537,6 +537,19 @@ void setIteratorRecordShape(CallExpr* call) {
   call->replace(shapeCall);
 }
 
+// Find the parent block that is either
+//  * a loop
+//  * or, a conditional
+// Ignore non-loop BlockStmts (these come up in particular with
+//  local and unlocal blocks).
+// (conditionals could probably be allowed if both sides yielded)
+static Expr* parentCondOrLoop(Expr* expr) {
+  Expr* parent = expr->parentExpr;
+  while (parent != NULL && !isLoopStmt(parent) && !isCondStmt(parent)) {
```

Comment:
I would prefer to explicitly ignore local/non-local/scopeless blocks instead of just discounting conditionals.

Reply:
I'll make it ensure it only skips BlockStmts (of one form or another).